Project/DAQ_System/Program/Src/app.cpp

```diff
@@ -117,22 +116,265 @@ void cppMain() {
-			CANMessageReceivedFlag = false;
+			canParseDataFrame();
+			if(CanID == 0x0CFFF048){
+				printf("\t\t 0x0CFFF048 \n");
+				printf("\r");
+				printf("RPM = %.2f ", DecodeDataFrame[0]);
+				printf("TPS = %.2f ", DecodeDataFrame[2]);
+				printf("Fuel Open Time = %.2f ", DecodeDataFrame[4]);
+				printf("Ignition Angle = %.2f \n", DecodeDataFrame[6]);
+				printf("\r");
+			}
+			if(CanID == 0x0CFFF548){
+				printf("\t\t 0x0CFFF548 \n");
+				printf("\r");
+				printf("Battery Volt = %.2f ", DecodeDataFrame[0]);
+				printf("Air Temp = %.2f ", DecodeDataFrame[2]);
+				printf("Coolant Temp = %.2f ", DecodeDataFrame[4]);
+				printf("Temp Type = %.2f \n", DecodeDataFrame[6]);
+				printf("\r");
+			}
 			__enable_irq();
 
 		}
 	}
 }
+
+
+void canFilterSetup(int FilterBankNum, uint32_t FilterExtID ){
+
+	// Filter initialization
+	CAN.FilterIdHigh = FilterExtID >> 13 ;
+	CAN.FilterIdLow = ((FilterExtID & 0x1FFF) <<3) | CAN_ID_EXT | CAN_RTR_DATA ;
+	CAN.FilterMaskIdHigh = 0xFFFF;
+	CAN.FilterMaskIdLow = 0xFFFF;
+	CAN.FilterFIFOAssignment = CAN_FILTER_FIFO0;
+	CAN.FilterBank = FilterBankNum;
+	CAN.FilterMode = CAN_FILTERMODE_IDMASK;
+	CAN.FilterScale = CAN_FILTERSCALE_32BIT;
+	CAN.FilterActivation = CAN_FILTER_ENABLE;
+	CAN.SlaveStartFilterBank = 17;
+	HAL_CAN_ConfigFilter(&hcan1,&CAN);
+}
+
+void canParseDataFrame(){
+
+	/*
+	 *
+	 the table below demonstrates the organization of data in each data type. This is specifically used for the canDecodeDataFrame() in which the data type of the particular
+	 id you are decoding is required.
+
+	 	 Data type | DLC | Data Breakdown | Length			data type | DLC | Data breakdown | length			Data type | DLC | Data Breakdown | Length
+	 	 -------------------------------------------	    -------------------------------------------			-------------------------------------------
+	 	 	 1	   |  8  |     1 - 2      | 2 bytes				 3	  |  8  | 	     1       | 1 byte			     4    |  7  |	  1 - 2 	 | 2 bytes
+	  	  	  	   |	 |     3 - 4 	  | 2 bytes				      |     |        2		 | 1 byte					  |     |     3 - 4		 | 2 bytes
+	  	  	  	   |	 |     5 - 6 	  | 2 bytes				      |     |        3		 | 1 byte					  |     |     5 - 6		 | 2 bytes
+	  	  	  	   |	 |     7 - 8 	  | 2 bytes				      |     |        4		 | 1 byte					  |     |     	7	 	 | 1 bytes
+	  	 --------------------------------------------			      |     |        5		 | 1 byte			--------------------------------------------
+	  	 	 2	   |  6  |     1 - 2      | 2 bytes				      |     |        6		 | 1 byte				 5    |  8  | 	  1 - 2      | 2 bytes
+	  	  	  	   |	 |     3 - 4 	  | 2 bytes				      |     |        7		 | 1 byte					  |     |     3 - 4      | 2 bytes
+	  	  	  	   |	 |     5 - 6 	  | 2 bytes				      |     |        8		 | 1 byte					  |		|       5        | 1 byte
+	  	 --------------------------------------------		------------------------------------------					  |		|		6		 | 1 byte
+																														  |		|       7 		 | 1 byte
```

Comment:
![CAN data ](https://github.com/DallasFormulaRacing/DataAcquisition2.0/assets/113472670/be868d8e-a2cf-4b15-936a-0a3bcebb70f2)


The formatting be kinda off when I did the commit so I just have a snippet of the code comment here. I wanted to keep this comment here since I understand that the data type I created is kinda confusing so I wanted some visual on the code to help make sense of it